### PR TITLE
add `doc_markdown` clippy lint

### PR DESCRIPTION
**Pull request by tcoratger**

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
all of these look correct and we actually need the `, for example [SealedHeader] doesn't get linked

https://reth.rs/docs/reth_testing_utils/generators/fn.random_header.html

please also take a look @shekhirin or @onbjerg

**Comment by tcoratger** [comment]
> all of these look correct and we actually need the `, for example [SealedHeader] doesn't get linked
> 
> https://reth.rs/docs/reth_testing_utils/generators/fn.random_header.html
> 
> please also take a look @shekhirin or @onbjerg

@mattsse Just fixed the case for [SealedHeader], I hope I didn't forget others

File: bin/reth/src/commands/db/tui.rs [hunk 1]
```diff
@@ -63,8 +63,8 @@ impl<T: Table> Entries<T> {
         }
     }
 
-    /// Sets the internal entries [Vec], converting the [Table::Value] into [RawValue<Table::Value>]
-    /// if needed.
+    /// Sets the internal entries [Vec], converting the [`Table::Value`] into
+    /// [`RawValue`<Table::Value>] if needed.
```
Comment (shekhirin) [inline bin/reth/src/commands/db/tui.rs:67]:
```suggestion
    /// [`RawValue<Table::Value>`] if needed.
```

**Review by shekhirin — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]
agree with this, did not know the lint existed

**Pull request merged**